src/docs/background/16-how-saving-payment-card-works-in-ftw/index.md

```diff
@@ -142,3 +142,162 @@ book a listing, there are couple of changes needed:
+
+```js
+=> sdk.transactions.initiate({ processAlias, transition: 'transition/request-payment', params: { listingId, bookingStart, bookingEnd, setupPaymentMethodForSaving: true }})
+```
+
+**When you use previously saved payment card**, the id of Stripe's
+payment method needs to be sent to Flex API as `paymentMethod`, when
+requesting payment.
+
+```js
+=> sdk.transactions.initiate({ processAlias, transition: 'transition/request-payment', params: { listingId, bookingStart, bookingEnd, paymentMethod: stripePaymentMethodId }})
+```
+
+> Note: params might be different in different transaction process
+> graphs.
+
+Check
+[`initiateOrder` thunk function](https://github.com/sharetribe/flex-template-web/blob/master/src/containers/CheckoutPage/CheckoutPage.duck.js#L171)
+and related
+[`orderParams`](https://github.com/sharetribe/flex-template-web/blob/master/src/containers/CheckoutPage/CheckoutPage.js#L373)
+from FTW.
+
+#### Step 2.
+
+```js
+=> stripe.handleCardPayment(stripePaymentIntentClientSecret, [card, paymentParams])
+```
+
+> Note: card (StripeElement), and paymentParams are not needed when
+> using previously saved payment card. FTW handles this in
+> [`handleCardPayment` thunk function](https://github.com/sharetribe/flex-template-web/blob/master/src/ducks/stripe.duck.js#L657).
+
+Stripe's frontent script checks if PaymentIntent needs extra actions
+from customer. Some payments might need Strong Customer Authentication
+(SCA). In practice, Stripe's script creates a popup (iframe) to card
+issuers site, where 3D secure v2 authentication flow can be completed.
+
+#### Step 3.
+
+```js
+=> sdk.transactions.transition({ id: transactionId, transition: 'transition/confirm-payment', params })
+```
+
+Inform Marketplace API, that PaymentIntent is ready to be captured after
+possible SCA authentication has been requested from user. FTW does that
+in
+[`confirmPayment` thunk call](https://github.com/sharetribe/flex-template-web/blob/master/src/containers/CheckoutPage/CheckoutPage.duck.js#L206)
+
+#### Step 4.
+
+```js
+=> sdk.messages.send({ transactionId: orderId, content: message })
+```
+
+FTW sends an initial message to transaction if customer has added a
+message.
+
+#### Step 5.
+
+As a final ste, we need to save the payment method, if customer has
+selected the "Save for later use" checkbox. So, this is relevant if user
+has selected onetime payment - instead of making a charge from the
+previously saved credit card.
+
+On FTW, we call
+[avePaymentMethod function](https://github.com/sharetribe/flex-template-web/blob/master/src/ducks/paymentMethods.duck.js#L200)
+that creates stripe customer and adds updates default payment method.
+
+There are 3 different scenarios, which require different calls to Flex
+API:
+
+**1. No StripeCustomer entity connected to Flex API:**
+
+```js
+=> sdk.stripeCustomer.create({ stripePaymentMethodId }, { expand: true, include: ['defaultPaymentMethod'] })
```

Comment:
Is `{ stripePaymentMethodId }` the same as `{ stripePaymentMethodId: stripePaymentMethodId }`?

Reply:
Yes